src/util/positioning.ts

```diff
@@ -214,42 +77,68 @@ export function positionElements(
     });
   }
 
-  // coordinates where to position
-
-  // Required for transform:
-  const style = targetElement.style;
-  style.position = 'absolute';
-  style.top = '0';
-  style.left = '0';
-  style['will-change'] = 'transform';
-
-  let testPlacement: Placement | null = null;
-  let isInViewport = false;
-  for (testPlacement of placementVals) {
-    let addedClasses = addClassesToTarget(testPlacement);
-
-    if (positionService.positionElements(hostElement, targetElement, testPlacement, appendToBody)) {
-      isInViewport = true;
-      break;
-    }
-
-    // Remove the baseClasses for further calculation
-    if (baseClass) {
-      addedClasses.forEach((classname) => { classList.remove(classname); });
-    }
-  }
+  const popperPlacements = placementVals.map((_placement) => { return convertPlacementToPopper(_placement); });
 
-  if (!isInViewport) {
-    // If nothing match, the first placement is the default one
-    testPlacement = placementVals[0];
-    addClassesToTarget(testPlacement);
-    positionService.positionElements(hostElement, targetElement, testPlacement, appendToBody);
-  }
+  let mainPlacement = popperPlacements.shift();
+
+  const bsModifier: Partial<Modifier<any, any>> = {
+    name: 'bootstrapClasses',
+    enabled: true,
```

Comment:
I think it should be `enabled: !!baseClass` to avoid executing it for components without the base class